client/src/pages/reschedule/RescheduleOrCancelAppointmentPage.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from "react";
+import { useParams, Redirect } from "react-router-dom";
+import { Grid, makeStyles, Paper, Divider } from "@material-ui/core";
+import axios from "axios";
+
+import Overview from "../../components/scheduler/Overview";
+import RescheduleCancelAppointment from "../../components/scheduler/RescheduleCancelAppointment";
+
+const RescheduleOrCancelAppointmentPage = ({ variant }) => {
+  const { appointmentId } = useParams();
+  const classes = useStyles();
+
+  const [appointmentDoesNotExist, setAppointmentDoesNotExist] = useState(false);
+  const [appointmentCancelled, setAppointmentCancelled] = useState(false);
+
+  const [appointmentDetails, setAppointmentDetails] = useState({
+    eventName: "",
+    duration: "",
+    eventUrl: "",
+    appointmentTime: "",
+    eventDescription: "",
+  });
+
+  useEffect(() => {
+    axios
+      .get(`/api/appointment/detail/${appointmentId}`)
+      .then((res) => {
+        // res.data: {eventUrl, duration, attendeeName, eventName, time, eventDescription}
+        setAppointmentDetails(res.data);
+      })
+      .catch(() => {
+        setAppointmentDoesNotExist(true);
+      });
+  }, [appointmentId]);
+
+  return (
+    <>
+      {appointmentCancelled && <Redirect to="/appointment/cancelled" />}
+      {appointmentDoesNotExist && <Redirect to="/appointment/does-not-exist" />}
```

Comment:
Could this instead redirect to the 404 page taras is working on?

Reply:
This page actually has another purpose, if the user try to return to the page after cancelling, it should tell them that the appointment has been cancelled rather than sending them back to an unrelated error page